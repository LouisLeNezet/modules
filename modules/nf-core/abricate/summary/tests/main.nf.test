nextflow_process {

    name "Test Process ABRICATE_SUMMARY"
    script "../main.nf"
    process "ABRICATE_SUMMARY"
    tag "modules"
    tag "modules_nfcore"
    tag "abricate"
    tag "abricate/summary"
    tag "abricate/run"

    test("Multiple reports - summary bacteroides_fragilis and haemophilus_influenzae") {

        setup {
            run("ABRICATE_RUN") {
                script "../../run/main.nf"
                process {
                    """
                    input[0] = Channel.fromList([
                        [
                            [ id:'test1' ],
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                        ],
                        [
                            [ id:'test2' ],
                            file(params.modules_testdata_base_path + 'genomics/prokaryotes/haemophilus_influenzae/genome/genome.fna.gz', checkIfExists: true)
                        ]
                    ])
                    input[1] = []
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = ABRICATE_RUN.out.report
                    .collect{ meta, report -> report }
                    .map{ report -> [ [ id: 'test_summary'], report ] }
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }

    test("Single report -- stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id: 'test_summary'], [] ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out,
                path(process.out.versions[0]).yaml
            ).match()
        }
    }
}
