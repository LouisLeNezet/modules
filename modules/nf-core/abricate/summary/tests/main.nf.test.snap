{
    "Multiple reports - summary bacteroides_fragilis and haemophilus_influenzae": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_summary"
                        },
                        "test_summary.txt:md5,a4ec7010e75404ce3a1033f0c4b4a7f9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1fadc3c51127e35186a8f1d4ec21ea07"
                ],
                "report": [
                    [
                        {
                            "id": "test_summary"
                        },
                        "test_summary.txt:md5,a4ec7010e75404ce3a1033f0c4b4a7f9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1fadc3c51127e35186a8f1d4ec21ea07"
                ]
            },
            {
                "ABRICATE_SUMMARY": {
                    "abricate": "1.0.1"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-10T22:53:20.811998373"
    },
    "Single report -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_summary"
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1fadc3c51127e35186a8f1d4ec21ea07"
                ],
                "report": [
                    [
                        {
                            "id": "test_summary"
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1fadc3c51127e35186a8f1d4ec21ea07"
                ]
            },
            {
                "ABRICATE_SUMMARY": {
                    "abricate": "1.0.1"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-10T22:53:27.087492986"
    }
}
